src/banyan_native/operations/prepare.rs

```diff
@@ -9,11 +16,106 @@ use std::{
     io::Read,
     path::{Path, PathBuf},
 };
-use tomb_common::{
-    blockstore::RootedBlockStore, metadata::FsMetadata, utils::wnfsio::path_to_segments,
-};
+use wnfs::private::PrivateNode;
+/// Given the input directory, the output directory, the manifest file, and other metadata,
+/// prepare the input directory into the output directory and store a record of how this
+/// operation was performed in the manifest file.
+///
+/// # Arguments
+///
+/// * `fs` - FileSystem to modify
+/// * `omni` - Context aware online / offline Drive
+/// * `client` - Means of connecting to the server if need be
+/// * `follow_links` - Whether or not to follow symlinks when bundling.
+///
+/// # Return Type
+/// Returns `Ok(())` on success, otherwise returns an error.
+pub async fn pipeline(
+    mut fs: FsMetadata,
+    omni: &mut OmniBucket,
+    client: &mut Client,
+    follow_links: bool,
+) -> Result<String, TombError> {
+    // Local is non-optional
+    let mut local = omni.get_local()?;
+
+    // If there is a remote Bucket with metadatas that include a content root cid which has already been persisted
+    if client.is_authenticated().await {
+        if let Ok(remote) = omni.get_remote() {
+            if let Ok(metadatas) = Metadata::read_all(remote.id, client).await {
+                if metadatas.iter().any(|metadata| {
+                    Some(metadata.root_cid.clone())
+                        == local.content.get_root().map(|cid| cid.to_string())
```

Comment:
( this is from previous PR; code moved )
this checks if there is already an upload with the root cid currently on disk. if and only if that is the case do we create a new 'delta'. this way, a delta represents all the content since we previously persisted data remotely